dogstatsd/Cargo.toml

```diff
@@ -0,0 +1,28 @@
+[package]
+name = "dogstatsd"
+rust-version.workspace = true
+edition.workspace = true
+version.workspace = true
+license.workspace = true
+
+[lib]
+bench = false
+
+[dependencies]
+datadog-protos = { version = "0.1.0", default-features = false, git = "https://github.com/DataDog/saluki-backport/" }
+ddsketch-agent = { version = "0.1.0", default-features = false, git = "https://github.com/DataDog/saluki-backport/" }
```

Comment:
We already have a ddsketch implementation in libdatadog, there is one in [mp-rs](https://github.com/DataDog/mp-rs/blob/020d8f72540ebfce2091ada1ed72e8eb5e97fd29/libs/dist_vocab/src/ddsketch.rs#L20) and there is [one on crates.io](https://crates.io/crates/sketches-ddsketch) that is not owned by DD. We should probably all agree on a single implementation.

Reply:
Interesting, this package uses the same implementation used by the Datadog Lambda Extension (owned by the Serverless team). Since I intend to use this DogStatsD package for a Serverless solution in Azure I'm inclined to stay consistent with the implementations.

https://github.com/DataDog/datadog-lambda-extension/tree/main/bottlecap/src/metrics

Reply:
I tried to remove the duplication but I read that this one is slightly different and optimized for the agent. So it might be not possible
https://github.com/DataDog/saluki/blob/main/lib/ddsketch-agent/src/lib.rs#L149
>>> /// This implementation is subtly different from the open-source implementations of `DDSketch`, as Datadog made some
/// slight tweaks to configuration values and in-memory layout to optimize it for insertion performance within the
/// agent.

Reply:
> slight tweaks to configuration values and in-memory layout to optimize it for insertion performance within the agent.

I'm not familiar with sketches, but optimizing in-memory layout and optimizing for insertion performance also sound relevant to a library, if it is feasible. Any insight as to whether it's appropriate or not?

Reply:
I am afraid I did not understand your question, can you elaborate more?

Reply:
👋🏻 

I'm the original author of the ported DDSketch implementation being used here. As the doc comments allude to, the main goal of the port was to maintain fidelity with the Agent's own DDSketch implementation, which at the time (and to this day) is not based on the open-source Go implementation of DDSketch.

I didn't write the original Agent-specific implementation, so I can't speak to all of the decisions made there, but I also tried to port over some of the code comments to, again, maintain fidelity with the Agent implementation.

While it probably behooves us, in a general sense, to have One True Implementation in Rust... maintaining compatibility with the Datadog Agent in the most important factor in this case.